WPMedia/fusion-news-theme-blocks
# Tmedia 468 identity block style button refactor pt1

## Description
Reusable button refactor only for identity blocks

## Jira Ticket
- [TMEDIA-468](https://arcpublishing.atlassian.net/browse/TMEDIA-468)

## Acceptance Criteria

Match button style criteria: 
Primary
Primary reverse
Secondary
Secondary reverse
Default - for when a type isn't passed

- Match mocks for identity blocks and styles: 
![Screen Shot 2021-09-22 at 11 18 23](https://user-images.githubusercontent.com/5950956/134382513-f367bc42-e383-419f-ad6d-d4936e730c72.png)
![Screen Shot 2021-09-22 at 11 18 34](https://user-images.githubusercontent.com/5950956/134382538-ed02c02a-0331-4997-9ac4-e2b06c0fbe5b.png)
![Screen Shot 2021-09-22 at 11 18 57](https://user-images.githubusercontent.com/5950956/134382581-c89753fe-93ed-4b94-8979-158d60632590.png)

## Test Steps

1. Checkout this branch `git checkout TMEDIA-468-identity-block-style-button-refactor-pt1`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/identity-block,@wpmedia/shared-styles`
3. Open pages for sign in with various primary color and nav bar background sites 
4. Go to mobile version by dragging each site
5. Sign in for a user and see the account button 
6. Drag in and see the mobile version of each site

demo-homepage: 
http://localhost/pf/demo-homepage/?_website=the-sun -> primary-color background
http://localhost/pf/demo-homepage/?_website=dagen -> light nav
http://localhost/pf/demo-homepage/?_website=the-prophet -> dark nav

## Effect Of Changes
### Before
- Using wrong color for icons in some cases
- Different naming for icons 

### After
<img width="1116" alt="Screen Shot 2021-09-20 at 10 11 28" src="https://user-images.githubusercontent.com/5950956/134382743-fd0e9908-17e7-48c5-ab49-e1d81a0bc4eb.png">
<img width="691" alt="Dagen" src="https://user-images.githubusercontent.com/5950956/134382767-d220626c-3eee-4a12-92d2-ddd13b2e7491.png">

## Dependencies or Side Effects
- Working on pt 2 for header nav buttons such as querly and sections
- Updated the form input to have the other button name updated

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
